openhcl/virt_mshv_vtl/src/cvm_cpuid/mod.rs

```diff
@@ -188,9 +192,45 @@ pub struct CpuidResults {
 type CpuidSubtable = HashMap<u32, CpuidResult>;
 
 /// Entry in [`CpuidResults`] for caching leaf value or its subleaves.
+#[derive(Inspect)]
+#[inspect(tag = "type")]
 enum CpuidEntry {
-    Leaf(CpuidResult),
-    Subtable(CpuidSubtable),
+    #[inspect(transparent)]
+    Leaf(#[inspect(with = "inspect_helpers::cpuid_result")] CpuidResult),
+    #[inspect(transparent)]
+    Subtable(#[inspect(with = "inspect_helpers::cpuid_subtable")] CpuidSubtable),
+}
+
+mod inspect_helpers {
+    use super::*;
+    use inspect::AsHex;
+    use inspect::Inspect;
+
+    pub(super) fn cpuid_result(result: &CpuidResult) -> impl Inspect + '_ {
+        inspect::adhoc(|req| {
+            req.respond()
+                .field("eax", AsHex(result.eax))
+                .field("ebx", AsHex(result.ebx))
+                .field("ecx", AsHex(result.ecx))
+                .field("edx", AsHex(result.edx));
+        })
+    }
+
+    pub(super) fn cpuid_table(table: &HashMap<CpuidFunction, CpuidEntry>) -> impl Inspect + '_ {
+        inspect::iter_by_key(
+            table
+                .iter()
+                .map(|(key, value)| (format!("{:?} ({:x})", key, key.0), value)),
```

Comment:
Can we just make the key the hex value in both cases? I don't want spaces and parens and open_enum identifiers in keys.

Reply:
yeah we can just make it the hex key, i found it useful to have both (but that's because i just started looking at cpuid so i don't have the idents memorized)